mypy/checkexpr.py

```diff
@@ -537,6 +537,25 @@ def check_typeddict_call_with_kwargs(self, callee: TypedDictType,
 
         return callee
 
+    def get_partial_self_var(self, expr: MemberExpr) -> Optional[Var]:
+        """Get variable node for a partial self attribute.
+
+        If the expression is not a self attribute, or attribute is not variable,
+        or variable is not partial, return None.
+        """
+        if not (isinstance(expr.expr, NameExpr) and
```

Comment:
Does this duplicate any logic used for `None` partial type handling? Could they be merged?

Reply:
> Does this duplicate any logic used for `None` partial type handling? Could they be merged?

Short answers are no and no. Some things may be however simplified after (or while) fixing https://github.com/python/mypy/issues/8043 (at least the first bullet).